back/dora/orientations/management/commands/close_expired_orientations.py

```diff
@@ -0,0 +1,84 @@
+from datetime import timedelta
+
+from django.conf import settings
+from django.core.management import BaseCommand
+from django.db.models import BooleanField, Case, Q, Value, When
+from django.utils import timezone
+
+from dora.orientations.emails import send_orientation_expiration_emails
+from dora.orientations.models import Orientation, OrientationStatus
+
+
+class Command(BaseCommand):
+    help = "Clôturer les orientations qui sont en cours  qui sont plus valables"
+
+    def handle(self, *args, **options):
+        self.stdout.write(
+            f"Clôture automatique des orientations en cours qui ne sont plus valables après {settings.ORIENTATION_EXPIRATION_PERIOD_DAYS} jours."
+        )
+
+        expiration_date = timezone.now() - timedelta(
+            days=settings.ORIENTATION_EXPIRATION_PERIOD_DAYS
+        )
+        email_cutoff_date = timezone.now() - timedelta(days=60)
+
+        expired_orientations = (
+            Orientation.objects.filter(
+                Q(
+                    processing_date__isnull=True,
+                    creation_date__lte=expiration_date,
+                )
+                | Q(
+                    processing_date__lte=expiration_date,
+                ),
+                status=OrientationStatus.PENDING,
+            )
+            .annotate(
+                should_send_email=Case(
+                    When(creation_date__gte=email_cutoff_date, then=Value(True)),
+                    default=Value(False),
+                    output_field=BooleanField(),
+                )
+            )
+            .select_related("service")
```

Comment:
```suggestion
            Orientation.objects.alias(last_state_changes_on=Coalesce("processing_date", "creation_date")).filter(
                last_state_changes_on__lte=expiration_date,
                status=OrientationStatus.PENDING,
            )
            .select_related("service")
```
`should_send_email` n'est pas utilisé dans la requête et ce n'est utilisé qu'une fois dans le script donc pas la peine de donner du boulot à la DB et autant le faire directement en python.

Je n'ai pas testé le changement avec `.alias()` mais normalement c'est fait pour donc ça devrait le faire, ça fait la même chose qu'avant mais je trouve que c'est plus simple à comprendre que d'avoir des conditions. Ça pourrais aussi être un `.annotate()` si tu veux remplacer le `start_date` après.